### DISCUSSION TITLE
开往改版方案v3

### DISCUSSION THREAD

**Discussion by HAHH9527** [travellings-link/travellings#1189] [发展建议]
# 前端改版
- [x] 独立前端库
- [x] 独立站点列表
- [x] 与站点列表分开部署互不影响
- [x] 从后端获取站点

# 审核改版
- [x] 审核不影响版本变动

# 新增后端
- [x] 后端获取站点编号，名称，url接口
- [x] 后端部署于服务器上
- [x] 通过pgsql同步服务器站点数据

**Comment by Jiaocz** [comment]
有一个想法，使用 Cloudflare Pages 的 [functions](https://developers.cloudflare.com/pages/platform/functions) 功能，实现边缘生成必要资源，避免返回所有 url 列表减少请求数量及请求体积，只需要单独的列表。
独立列表也可以使用 Cloudflare Pages 支持的 Workers KV / R2 存储，使用 functions 函数进行维护等。

## Pros
- 无需服务器
- 边缘计算可在CDN环境进行生成，免去CDN到服务器的路由时间
- 可以使用 Cloudflare Pages 各种预设 Functions
- 也许可以增加自助提交审核的功能

## Cons
- 审核人员认证部分可能较复杂
- Pages 目前 Functions 调用次数限制 100k/day （可填表增加）
- 可能不方便在 Github 仓库这样的地方展示所有成员信息
- 冷启动可能会慢

其他信息待研究补充

**Reply by HAHH9527** [reply]
是否可以审核依旧在GitHub上完成，依据特定文件生成 functions 所需配置

**Reply by ghost** [reply]
友情提示，云函数很多坑（之前弄[Cloud Functions](https://cloud.google.com/functions)全是坑

**Reply by Jiaocz** [reply]
> 是否可以审核依旧在GitHub上完成，依据特定文件生成 functions 所需配置

可以，办法总比困难多

**Reply by Jiaocz** [reply]
> 友情提示，云函数很多坑（之前弄[Cloud Functions](https://cloud.google.com/functions)全是坑

确实，functions现在也还是beta，昨天刚刚发现一个bug提交了issue，不过最简单的功能用一用也没什么，比如仅仅随机返回一个成员信息还是没什么问题的

**Reply by Bluemangoo** [reply]
> 友情提示，云函数很多坑（之前弄[Cloud Functions](https://cloud.google.com/functions)全是坑

我之前写vercel的云函数感觉还行，应该是一样的。

反正都是写node函数，写出来应该容易，可以试试的。

**Comment by jiangyangcreate** [comment]
已全部采纳并完成
